checker.go

```diff
@@ -107,6 +109,42 @@ func (c *cmpEqualsChecker) Check(got interface{}, args []interface{}, note func(
 //
 var DeepEquals = CmpEquals()
 
+// ContentEquals is a Checker checking that the two provided slices have the
+// same items, without respect to order.
+// For instance:
+//
+//     c.Assert([]int{1, 2, 3], qt.ContentEquals, []int{3, 2, 1})
+//
+var ContentEquals Checker = &contentEqualsChecker{
```

Comment:
How about just:

    // ContentEquals is like DeepEquals but any slices in the compared values
    // will be sorted before being compared.
    var ContentEquals = CmpEquals(cmpopts.SortSlices(func(x, y interface{}) bool {
		// TODO frankban: implement a proper sort function.
		return pretty.Sprint(x) < pretty.Sprint(y)
    }))

?

That is, I'm not sure there's a particular reason to check that the top level values are slices, and the logic you currently have sorts all slices in the values, not just the top level, and I think that's probably fine.

Reply:
Also, having it as a simple CmpEquals checker perhaps also acts as a nice example of how CmpEquals can be used to build more sophisticated DeepEquals-style checkers quite easily.

Reply:
Done.